libsel4vmmplatsupport/src/plat/zynqmp/devices/vuart.c

```diff
@@ -0,0 +1,397 @@
+/*
```

Comment:
Instead of vuart.c, would it be easier to name this file based on which device it is emulating? eg virt-cdns-uart-r1p12.c or whatever the device compatibility string is for the device that this emulates?

Reply:
That would be good behavior. I'm honestly thinking at this point that the `vuart` be refactored out of this PR and into its own, since the vuart isn't technically necessary (passthrough serial and virtio-console both work well)